src/Kaixa.groovy

```diff
@@ -2321,31 +2329,74 @@ public class Kaixa {
       cachedPasswords.put(name, password);
     }
 
-    // Wait for discovery page to load
-    Kaixa.waitForElementPresent('#idp_1001962798_button');
-    Kaixa.click('#idp_1001962798_button');
-
-    // Wait for the page to load
-    Kaixa.waitForElementVisible('#username');
-    Kaixa.waitForElementVisible('#password');
-
-    // Add credentials
-    Kaixa.typeInto('#username', username);
-    Kaixa.typeInto('#password', password);
-
-    // Click "submit"
-    Kaixa.click('.btn-primary');
+		boolean isHarvardKey  = obj.has('isHarvardKey') && obj.getBoolean('isHarvardKey');
+		boolean needs2FA = isHarvardKey && obj.has('needs2FA') && obj.getBoolean('needs2FA');
+
```

Comment:
In GlobalCredentials, the user will be something like:
```
	public static String admin ='''
		{
		   'username': 'my_name@harvard.edu',
		   'password': '',
           'isHarvardKey': true,
           'needs2FA': true,
		}

```
I'm not attached to specific names so please suggest better ones.

Reply:
So now:
public static String admin ='''
		{
		   'username': 'guest_name@outlook.com',
		   'password': 'blah',
           'isHarvardGuest': true,
		}